src/util.cc

```diff
@@ -825,6 +825,36 @@ double GetLoadAverage() {
 }
 #else
 double GetLoadAverage() {
+# if defined(__linux__)
+  // Only try to open the file once.
+  static int fd = -1;
+  if (fd == -1) {
+    fd = open("/proc/loadavg", O_RDONLY);
+    if (fd < 0) {
+      fd = -2;
+    }
+  }
+
+  if (fd >= 0) {
+    char buf[65], *p;
+    ssize_t nbr = pread(fd, buf, sizeof(buf) - 1, 0);
+    // Load average should be at least "1 5 9 n/".
+    if (nbr < 8) {
+      return -0.0f;
+    }
```

Comment:
If `nbr < 8` we should just assume there is something wrong and let the non linux code try to get the load average.

Reply:
Agreed, I will change this.